main.nf

```diff
@@ -447,31 +475,59 @@ process bwa {
 /*
  * STEP 3.2 - post-alignment processing
  */
+if(!params.ATACseq){
+    process samtools {
+        tag "${bam.baseName}"
+        publishDir path: "${params.outdir}/bwa", mode: 'copy',
+                   saveAs: { filename ->
+                       if (filename.indexOf(".stats.txt") > 0) "stats/$filename"
+                       else params.saveAlignedIntermediates ? filename : null
+                   }
 
-process samtools {
-    tag "${bam.baseName}"
-    publishDir path: "${params.outdir}/bwa", mode: 'copy',
-               saveAs: { filename ->
-                   if (filename.indexOf(".stats.txt") > 0) "stats/$filename"
-                   else params.saveAlignedIntermediates ? filename : null
-               }
+        input:
+        file bam from bwa_bam
 
-    input:
-    file bam from bwa_bam
+        output:
+        file '*.sorted.bam' into bam_picard, bam_for_mapped
+        file '*.sorted.bam.bai' into bai_picard, bai_for_mapped
+        file '*.sorted.bed' into bed_total
+        file '*.stats.txt' into samtools_stats
 
-    output:
-    file '*.sorted.bam' into bam_picard, bam_for_mapped
-    file '*.sorted.bam.bai' into bai_picard, bai_for_mapped
-    file '*.sorted.bed' into bed_total
-    file '*.stats.txt' into samtools_stats
+        script:
+        """
+        samtools sort $bam -o ${bam.baseName}.sorted.bam
+        samtools index ${bam.baseName}.sorted.bam
+        bedtools bamtobed -i ${bam.baseName}.sorted.bam | sort -k 1,1 -k 2,2n -k 3,3n -k 6,6 > ${bam.baseName}.sorted.bed
+        samtools stats ${bam.baseName}.sorted.bam > ${bam.baseName}.stats.txt
+        """
+    }
+} else {
+    process samtools {
+        tag "${bam.baseName}"
+        publishDir path: "${params.outdir}/bwa", mode: 'copy',
+                   saveAs: { filename ->
+                       if (filename.indexOf(".stats.txt") > 0) "stats/$filename"
+                       else params.saveAlignedIntermediates ? filename : null
+                   }
 
-    script:
-    """
-    samtools sort $bam -o ${bam.baseName}.sorted.bam
-    samtools index ${bam.baseName}.sorted.bam
-    bedtools bamtobed -i ${bam.baseName}.sorted.bam | sort -k 1,1 -k 2,2n -k 3,3n -k 6,6 > ${bam.baseName}.sorted.bed
-    samtools stats ${bam.baseName}.sorted.bam > ${bam.baseName}.stats.txt
-    """
+        input:
+        file bam from bwa_bam
+        file genomeSizeTable from genomeSizeTable
+
+        output:
+        file '*.sorted.bam' into bam_picard, bam_for_mapped
+        file '*.sorted.bam.bai' into bai_picard, bai_for_mapped
+        file '*.sorted.bed' into bed_total
+        file '*.stats.txt' into samtools_stats
+
+        script:
+        """
+        samtools sort $bam | bedtools bamtobed | awk -v OFS="\\t" '{if(\$6=="+"){\$2+=4;\$3+=4;print} else if(\$6=="-" && \$2>5){\$2-=5;\$3-=5;print} else if(\$6=="-" && \$2<=5){\$3=\$3-\$2+1;\$2=1;print}}' | sort -k 1,1 -k 2,2n -k 3,3n -k 6,6 > ${bam.baseName}.sorted.bed
```

Comment:
Wow! What is happening here?

It looks like the inputs and outputs are the same, so you can reduce code duplication by putting the if block inside the `script` instead.

Reply:
Because the file `genomeSizeTable` does not exist for the case when no `--ATACseq` is specified.

Reply:
Yeah, because we'd need to always make the table - I think that would be fine to do. But maybe not worth the effort to change.

I'm still not sure what the commands are doing here though?